src/parser/jobs/mnk/modules/RiddleOfFire.tsx

```diff
@@ -1,198 +1,257 @@
-			</Trans>,
-		}))
-	}
+	buffStatus = this.data.statuses.RIDDLE_OF_FIRE
 
-	override output() {
-		return <RotationTable
-			targets={[
+	override initialise() {
+		super.initialise()
+
+		const suggestionWindowName = <ActionLink action="RIDDLE_OF_FIRE"/>
+
+		this.addEvaluator(new MasterfulBlitzEvaluator(this.blitzSkills, this.data.actions.CELESTIAL_REVOLUTION, this.data.actions.MASTERFUL_BLITZ, this.data.actions.BROTHERHOOD))
+		this.addEvaluator(new BrotherhoodEvaluator(this.data.actions.BROTHERHOOD))
+		this.addEvaluator(new ExpectedGcdCountEvaluator({
+			expectedGcds: EXPECTED_GCDS,
+			globalCooldown: this.globalCooldown,
+			suggestionIcon: this.data.actions.RIDDLE_OF_FIRE.icon,
+			severityTiers: TOTAL_GCD_DURING_ROF_SEVERITY,
+			suggestionWindowName: suggestionWindowName,
+			suggestionContent: <Trans id="mnk.rof.suggestions.gcd.content">Aim to hit {EXPECTED_GCDS} GCDs during
+				each <ActionLink action="RIDDLE_OF_FIRE"/> window.</Trans>,
+			adjustCount: window => {
+				// 6SS counts as 2 GCDs
+				return -this.numberOfSixSidedStarInWindow(window)
+			},
+		}))
+		this.addEvaluator(new LimitedActionsEvaluator({
+			expectedActions: [
 				{
-					header: <Trans id="mnk.rof.table.header.gcds">GCDs</Trans>,
-					accessor: 'gcds',
+					action: this.data.actions.MEDITATION,
+					expectedPerWindow: 0,
 				},
 				{
-					header: <DataLink showName={false} action="THE_FORBIDDEN_CHAKRA"/>,
-					accessor: 'forbiddenChakra',
+					action: this.data.actions.ANATMAN,
+					expectedPerWindow: 0,
 				},
 				{
-					header: <DataLink showName={false} action="MASTERFUL_BLITZ"/>,
-					accessor: 'blitzes',
+					action: this.data.actions.FORM_SHIFT,
+					expectedPerWindow: 0,
 				},
-			]}
-			data={this.history
-				.map(riddle => ({
-					start: riddle.start - this.parser.pull.timestamp,
-					end: riddle.end != null ?
-						riddle.end - this.parser.pull.timestamp
-						: riddle.start - this.parser.pull.timestamp,
-					targetsData: {
-						gcds: {
-							actual: riddle.gcds - riddle.meditations,
-							expected: EXPECTED.GCDS - riddle.stars,
-						},
-						forbiddenChakra: {
-							actual: riddle.chakras,
-						},
-						blitzes: {
-							actual: riddle.casts.filter(event => this.blitzActions.includes(event.action)).length,
-							expected: EXPECTED.BLITZES,
-						},
-					},
-					rotation: riddle.casts,
-				}))
-			}
-			onGoto={this.timeline.show}
-		/>
+			],
+			suggestionIcon: this.data.actions.MEDITATION.icon,
+			suggestionContent: <Trans id="mnk.rof.suggestions.meditation.content">
+				Avoid using <ActionLink action="MEDITATION"/> under <ActionLink action="RIDDLE_OF_FIRE"/> as this is
+				essentially wasting a GCD.
+			</Trans>,
+			suggestionWindowName,
+			severityTiers: BAD_GCD_DURING_ROF_SEVERITY,
+		}))
+	}
+
+	private numberOfSixSidedStarInWindow(window: HistoryEntry<EvaluatedAction[]>): number {
+		return window.data.filter(value => value.action.id === this.data.actions.SIX_SIDED_STAR.id).length
 	}
```

Comment:
This is only called once and it's a one-liner, it doesn't need to be in its own function. You do this in a few places (e.g. `checkIfBrotherhoodIsInWindow` - someone trying to contribute to this file in the future is going to have to jump around a lot to follow the logic.